### ISSUE TITLE
Refactor RewardFn

### ISSUE THREAD

**Issue by AdamGleave** [HumanCompatibleAI/imitation#427]
From ernestum's suggestions in https://github.com/HumanCompatibleAI/imitation/pull/426:

> I also looked a bit into the rewards corner of the lib and I would propose the following refactoring (which would also get rid of common.py):
> 
> - Turn RewardFn into an abstract class. Then it does not just define the signature of a reward function but also the argument names and a docstring explaining the semantics of the function call.
> - Rename common.py -> reward_fn.py
> - Make all the reward networks inherit from RewardFn. Turn their .predict() into __call__() (or make the __call__() execute predict

**Comment by AdamGleave** [comment]
I agree with the suggestion to turn `RewardFn` into an ABC. But I think having `RewardNet` subclass it won't work as they already inherit an incompatible `__call__`  from nn.Module` (that runs hooks then calls `forward`). 

So I think we can just do the first two points but not the third, WDYT @ernestum?

**Comment by AdamGleave** [comment]
Fixed in https://github.com/HumanCompatibleAI/imitation/pull/478
